eliot/tests/test_output.py

```diff
@@ -503,6 +511,78 @@ def raiser(i):
                               (key, value) in message.items()))
 
 
+    def test_destinationExceptionCaught(self):
+        """
+        If a destination throws an exception, an appropriate error is
+        logged.
+        """
+        logger = Logger()
+        logger._destinations = Destinations()
+        dest = BadDestination()
+        logger._destinations.add(dest)
+
+        message = {"hello": 123}
+        logger.write({"hello": 123})
+        assertContainsFields(
+            self, dest[0],
+            {"message_type": "eliot:destination_failure",
+             "message": logger._safeUnicodeDictionary(message),
+             "reason": "ono",
+             "exception": "eliot.tests.test_output.MyException"})
+
+
+    def test_destinationMultipleExceptionsCaught(self):
+        """
+        If multiple destinations throw an exception, an appropriate error is
+        logged for each.
+        """
+        logger = Logger()
+        logger._destinations = Destinations()
+        logger._destinations.add(BadDestination())
+        logger._destinations.add(lambda msg: 1/0)
+        messages = []
+        logger._destinations.add(messages.append)
+
+        try:
+            1/0
+        except ZeroDivisionError as e:
+            zero_divide = str(e)
+        zero_type = ZeroDivisionError.__module__ + ".ZeroDivisionError"
+
+        message = {"hello": 123}
+        logger.write({"hello": 123})
+        self.assertEqual(
+            messages,
```

Comment:
Ah, the penny drops! So exceptions raised in failing destinations are logged to the working destinations.
Perhaps it'd be nice to demonstrate that the order of the destinations is unimportant ie failures in  destinations further down the list will be logged to working destinations earlier in the destination list.